### PR TITLE
[#10465] Instructor enroll students: error message should be a modal

### LINKED ISSUE #10465: Instructor enroll students: error message should be a modal
I think the enroll page is one place where we have to show error messages as modals and not as toasts. Those messages contain vital information that need to persist until the user is done with them.

### PR DESCRIPTION
**Pull request by t-cheepeng**
Fixes #10465, #10038 

**Outline of Solution**

erere

- Added in card for long error message

### CODE REVIEW

**Comment by t-cheepeng** [comment]
@damithc Now, only the very long error message persists and is dismissable with the `X`. Do you want the other shorter error messages to also persist here?(E.g. Student name cannot be empty, etc...). Don't see a point in these messages persistence since they are very short. One sentence max.

**Comment by damithc** [comment]
> @damithc Now, only the very long error message persists and is dismissable with the `X`. Do you want the other shorter error messages to also persist here?(E.g. Student name cannot be empty, etc...). Diden't see a point in persisting these messages since they are very short where they are at most one sentence.

I think best to persist all error messages in this page. Users need time to correct the error (they may need to refer back to the error message later when they retry) and sometimes even to take screenshot and send back to us.

**Comment by t-cheepeng** [comment]
>I think best to persist all error messages in this page

Updated the screenshots. All error messages from enrollment persist now

File: src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div *ngIf="coursePresent">
     <h1>Enroll Students for {{ courseId }}</h1>
     <tm-loading-retry [shouldShowRetry]="hasLoadingStudentsFailed" [message]="'Failed to load students'" (retryEvent)="getCourseEnrollPageData(courseId)">
-      <div *ngIf='!hasLoadingStudentsFailed' class="card card-primary">
+      <div *ngIf='!hasLoadingStudentsFailed'class="card card-primary">
```
Comment (madanalogy) [inline src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html:5]: Why was the space removed?

**Comment by t-cheepeng** [comment]
> Possible to just remove the error toast for those cases?

Removed and updated screencaps above

File: src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html [hunk 2]
```diff
@@ -20,7 +20,7 @@ <h1>Enroll Students for {{ courseId }}</h1>
             <tm-status-message [messages]="statusMessage"></tm-status-message>
             <div class="card card-default">
               <div id="toggle-existing-students" class="card-header cursor-pointer"
-                   (click)="toggleExistingStudentsPanel()">
+               (click)="toggleExistingStudentsPanel()">
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html:23]: Unrelated change here

File: src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html [hunk 3]
```diff
@@ -49,6 +49,14 @@ <h1>Enroll Students for {{ courseId }}</h1>
                   [hidden]="isExistingStudentsPanelCollapsed" @collapseAnim>
               </hot-table>
             </div>
+
+            <div *ngIf="showEnrollErrorMessage && enrollErrorMessage" class="card card-default">
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.html:53]: Is there a case whereby `enrollErrorMessage` is non-empty but `showEnrollErrorMessage` is `false`? If not, the condition `showEnrollErrorMessage` is actually redundant.

**Comment by damithc** [comment]
> Updated the screenshots. All error messages from enrollment persist now

👍  Good work @t-cheepeng
